crates/bevy_animation/src/lib.rs

```diff
@@ -7,24 +7,27 @@
 
 //! Animation for the game engine Bevy
 
+extern crate alloc;
+
 pub mod animatable;
 pub mod graph;
 pub mod transition;
 mod util;
 
-use std::cell::RefCell;
-use std::collections::BTreeMap;
-use std::hash::{Hash, Hasher};
-use std::iter;
-use std::ops::{Add, Mul};
+use alloc::collections::BTreeMap;
+use core::{
+    cell::RefCell,
+    hash::{Hash, Hasher},
+    iter,
+    ops::{Add, Mul},
+};
 
 use bevy_app::{App, Plugin, PostUpdate};
 use bevy_asset::{Asset, AssetApp, Assets, Handle};
 use bevy_core::Name;
 use bevy_ecs::{entity::MapEntities, prelude::*, reflect::ReflectMapEntities};
 use bevy_math::{FloatExt, FloatPow, Quat, Vec3};
-use bevy_reflect::std_traits::ReflectDefault;
-use bevy_reflect::Reflect;
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
```

Comment:
Due to the use of `cargo +nightly fmt` to split and re-merge `use` statements, there are some additional changes beyond just the `std`, `alloc`, and `core` crates. It has been indicated in the `rustfmt.toml` file that _ideally_ we would already be using `imports_granularity = "Crate"` if it was a stable feature, so I don't consider this controversial.